src/GL/rsp_gl.S

```diff
@@ -685,34 +746,17 @@ GLCmd_PreInitPipe:
     or modes0, t7
 2:
 
-    # Combiner
-    move t5, is_points
-    lw t3, %lo(GL_STATE_TEX_ENV_MODE)
-
-    li t4, GL_REPLACE
-    bne t3, t4, 1f
+    j rdpq_skiptexture
     nop
-    ori t5, 0x2
-1:
+rdpq_mergemask:
+    # If we have GL_RDPQ_TEXTURING_N64 active, we need to use some of the RDPQ params instead of GL's
 
-    beqz active_tex, 1f
-    lw t0, TEXTURE_FLAGS_OFFSET(active_tex)
-    andi t0, TEX_FLAG_COMPLETE
-    srl t0, 1
-    or t5, t0
-1:
-    sll t5, 3
-    lw t0, %lo(COMBINER_TABLE) + 0x0(t5)
-    lw t1, %lo(COMBINER_TABLE) + 0x4(t5)
-    lw t2, %lo(COMBINER_MIPMAPMASK_TABLE) + 0x0(t5)
-    lw t3, %lo(COMBINER_MIPMAPMASK_TABLE) + 0x4(t5)
-
-    # TODO: The following is sort of equivalent to RDPQCmd_ResetMode. Maybe make that callable from ucode?
+    lw t1, %lo(RDPQ_OTHER_MODES) + 0x0
+    and t1, RDPQ_TEXTURING_MASK
+    and t2, modes0, ~(RDPQ_TEXTURING_MASK)
+    or  modes0, t1, t2
 
-    sw t0, %lo(RDPQ_COMBINER) + 0x0
-    sw t1, %lo(RDPQ_COMBINER) + 0x4
-    sw t2, %lo(RDPQ_COMBINER_MIPMAPMASK) + 0x0
-    sw t3, %lo(RDPQ_COMBINER_MIPMAPMASK) + 0x4
+rdpq_skiptexture:
```

Comment:
Same as above, this label should also use the `gl_` prefix. Additionally, I think the name is somewhat misleading altogether. Maybe it should be `gl_save_modes` or something?
```suggestion
gl_save_modes:
```